x/evm/abci.go

```diff
@@ -166,6 +195,25 @@ func handleContractCallWithToken(ctx sdk.Context, event types.Event, bk types.Ba
 	return nil
 }
 
+func setGeneralMessageToNexus(ctx sdk.Context, n types.Nexus, event types.Event) error {
+	e := event.GetContractCall()
+	if e == nil {
+		panic(fmt.Errorf("event is nil"))
+	}
+
+	message := nexus.NewGeneralMessage(
+		string(event.GetID()),
```

Comment:
need a way to link the counter id with the event id for execution service

Reply:
`setMessage/EnqueueMessage` can return the `id` and we can emit an event to link the two? Otherwise we'd need a separate counter anyways in axelarnet when sending messages from Cosmos side.

Although, I agree this simplifies microservice work if we keep it as is for evm txs at least.

Reply:
discussed to leave this as-is for now for simplicity